pkg/apiserver/apiserver.go

```diff
@@ -102,18 +105,50 @@ func NewAPIServer(clientConfig *rest.Config) (*APIServer, error) {
 		return nil, err
 	}
 
-	return &APIServer{server, make(chan struct{})}, nil
+	return &APIServer{server, make(chan struct{}), aggClient}, nil
 }
 
 // Spawns go routine that exits when apiserver is stopped
-func (as *APIServer) Run() {
+func (as *APIServer) Run() error {
+	const (
+		startupTimeout = 30 * time.Second
+	)
 	go as.server.PrepareRun().Run(as.stopCh)
+
+	// wait for apiservice to have healthy status
+	timeout := time.Now().Add(startupTimeout)
+	for time.Now().Before(timeout) {
```

Comment:
i just noticed that this a hot loop (will make calls one after another). i think it might be worth switching this to a simple i++ loop doing <30 times, and time.Sleep(1) in beetween.

Reply:
Ah yea, good call

Reply:
Updated